src/lib/rng/rng.cpp

```diff
@@ -8,36 +8,31 @@
 
 #include <botan/entropy_src.h>
 #include <botan/internal/loadstor.h>
-#include <botan/internal/os_utils.h>
 
 #if defined(BOTAN_HAS_SYSTEM_RNG)
    #include <botan/system_rng.h>
 #endif
 
+#if defined(BOTAN_HAS_OS_UTILS)
+   #include <botan/internal/os_utils.h>
+#endif
+
 #include <array>
 
 namespace Botan {
 
 void RandomNumberGenerator::randomize_with_ts_input(std::span<uint8_t> output) {
    if(this->accepts_input()) {
-      constexpr auto s_hd_clk = sizeof(decltype(OS::get_high_resolution_clock()));
-      constexpr auto s_sys_ts = sizeof(decltype(OS::get_system_timestamp_ns()));
-      constexpr auto s_pid = sizeof(decltype(OS::get_process_id()));
-
-      std::array<uint8_t, s_hd_clk + s_sys_ts + s_pid> additional_input = {0};
-      auto s_additional_input = std::span(additional_input.begin(), additional_input.end());
-
-      store_le(OS::get_high_resolution_clock(), s_additional_input.data());
-      s_additional_input = s_additional_input.subspan(s_hd_clk);
+      std::array<uint8_t, 32> additional_input = {0};
 
 #if defined(BOTAN_HAS_SYSTEM_RNG)
-      System_RNG system_rng;
-      system_rng.randomize(s_additional_input);
-#else
-      store_le(OS::get_system_timestamp_ns(), s_additional_input.data());
-      s_additional_input = s_additional_input.subspan(s_sys_ts);
+      system_rng().randomize(additional_input);
+#endif
 
-      store_le(OS::get_process_id(), s_additional_input.data());
+#if defined(BOTAN_HAS_OS_UTILS)
+      store_le(&additional_input[0], OS::get_high_resolution_clock());
+      store_le(&additional_input[8], OS::get_system_timestamp_ns());
+      store_le(&additional_input[16], OS::get_process_id());
 #endif
 
       this->fill_bytes_with_input(output, additional_input);
```

Comment:
For the record: This will typically* overwrite the majority of the random bits with the high-resolution timestamps below. Not sure its worth the extra complexity to handle that more efficiently, but perhaps like this:

```C++
      std::array<uint8_t, 32> additional_input = {0};

#if defined(BOTAN_HAS_OS_UTILS)
      store_le(std::span{additional_input}.subspan<0, 8>(), OS::get_high_resolution_clock());
      store_le(std::span{additional_input}.subspan<8, 8>(), OS::get_system_timestamp_ns());
      store_le(std::span{additional_input}.subspan<16, 4>(), OS::get_process_id());
      auto randomize = std::span{additional_input}.subspan<20>();
#else
      auto randomize = std::span{additional_input};
#endif

#if defined(BOTAN_HAS_SYSTEM_RNG)
      system_rng().randomize(randomize);
#else
      BOTAN_UNUSED(randomize);
#endif

      this->fill_bytes_with_input(output, additional_input);
```

\* when both system rng and OS utils are available

Reply:
To me it seemed easier to just accept this and always perform a longer read but it is a bit subtle and lacking at least a comment might strike someone as a bug/unintentional. Your approach is probably better here.